pkgs/top-level/default.nix

```diff
@@ -77,14 +73,19 @@ let
     };
 
   trivialBuilders = self: super:
-    (import ../build-support/trivial-builders.nix {
+    import ../build-support/trivial-builders.nix {
       inherit lib; inherit (self) stdenv stdenvNoCC; inherit (self.xorg) lndir;
-    });
+    };
 
-  stdenvDefault = self: super: (import ./stdenv.nix topLevelArguments) pkgs;
+  stdenvDefault = self: super:
+    import ./stdenv.nix {
+      inherit system bootStdenv crossSystem config platform lib nixpkgsFun pkgs;
+    };
 
   allPackages = self: super:
-    let res = import ./all-packages.nix topLevelArguments res self;
+    let res = import ./all-packages.nix
+          { inherit system noSysDirs config crossSystem platform lib nixpkgsFun; }
+          res self;
```

Comment:
Hmm it's supposed to be beginning of `res` + 2, but that does look odd. Seeing that
```nix
let # note newline
  foo = ...;
  bar = ...;
```
would be IMO proper style for multiple lets, I'll just unindent back to two from `let`.